pkg/pillar/cmd/agent.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"flag"
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// AgentBase - Interface for all agents to use the AgentBase.
+// Each agent is expected to have a Context that implements
+// this interface.
+type AgentBase interface {
+	AgentBaseContext() *AgentBaseContext
+	AddAgentSpecificCLIFlags()
+	ProcessAgentSpecificCLIFlags()
+}
+
+// AgentBaseCLIParams - stores all the common cli params
+type AgentBaseCLIParams struct {
+	DebugOverride bool
+}
+
+// AgentBaseContext - a struct that represents a general agent context
+type AgentBaseContext struct {
+	CLIParams   AgentBaseCLIParams
+	ErrorTime   time.Duration
+	WarningTime time.Duration
+	AgentName   string
+}
+
+// processCLIFlags - Add flags common to all agents
+func processCLIFlags(agentBase AgentBase) {
+	debugPtr := flag.Bool("d", false, "Debug flag")
+	agentBase.AddAgentSpecificCLIFlags()
+	flag.Parse()
+	ctx := agentBase.AgentBaseContext()
+	ctx.CLIParams.DebugOverride = *debugPtr
+	agentBase.ProcessAgentSpecificCLIFlags()
+}
+
+// Run - a general run function that will handle all of the common code for agents
+func Run(agentSpecificContext AgentBase) {
+	processCLIFlags(agentSpecificContext)
+	ctx := agentSpecificContext.AgentBaseContext()
+	debugOverride := ctx.CLIParams.DebugOverride
+	if debugOverride {
+		log.SetLevel(log.DebugLevel)
+	} else {
+		log.SetLevel(log.InfoLevel)
+	}
+	err := agentlog.Init(ctx.AgentName, "")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := pidfile.CheckAndCreatePidfile(ctx.AgentName); err != nil {
+		log.Fatal(err)
```

Comment:
It's not nice by a package to call log.Fatal. Better to return err and have the caller do the log.Fatal.